app/bundles/FormBundle/Tests/Controller/SubmissionFunctionalTest.php

```diff
@@ -26,6 +26,88 @@ final class SubmissionFunctionalTest extends MauticMysqlTestCase
 {
     protected $useCleanupRollback = false;
 
+    public function testRedirectPostAction(): void
+    {
+        // Create the test page via API.
+        $payload = [
+            'isPublished'  => true,
+            'title'        => 'Test',
+            'alias'        => 'test-form-redirect-target-page',
+            'language'     => 'en',
+            'redirectType' => null,
+            'redirectUrl'  => null,
+            'template'     => 'blank',
+            'customHtml'   => '<!DOCTYPE html><html><head></head><body>Test</body></html>',
+        ];
+
+        $this->client->request(Request::METHOD_POST, '/api/pages/new', $payload);
+        $clientResponse = $this->client->getResponse();
+
+        $this->assertSame(Response::HTTP_CREATED, $clientResponse->getStatusCode(), $clientResponse->getContent());
+
+        $response = json_decode($clientResponse->getContent(), true);
+        $pageId   = $response['page']['id'];
```

Comment:
Just an FYI for future PRs, it's more performant to create entities that are necessary for the functional test with entity manager. And it is also more convenient to use
```suggestion
        $page = new Page();
        $page->setTitle('Test');
        $page->setAlias('test-form-redirect-target-page');
        $page->setCustomHtml('<!DOCTYPE html><html><head></head><body>Test</body></html>');
        $this->em->persist($page);
        $this->em->flush();
        $pageId = $page->getId();
```
We used to do a refactoring earlier to speed up the test suite and this was one of the techniques that helped a lot.

Reply:
Ah, okay. I saw the "API call method" used at so many places I looked. So, I thought this was the way to go.

TBH I found the API method very inconvenient as you need to figure out the correct JSON payload, first :upside_down_face: 

Fits the link you posted, yesterday, WRT the code quality part :sweat_smile: